### PR TITLE
add openai model endpoint wrapper

### PR DESCRIPTION
**Pull request by imatiach-msft**
Add openai model endpoint wrapper.  This includes the wrapper around an azure openai API, the retry logic to attempt calling the API multiple times incase of network flakiness, and logic to re-wrap the model's output as a pandas dataframe.

### CODE REVIEW

File: python/ml_wrappers/model/openai_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        self.messages = messages
+
+    def fetch(self):
+        """Call the openai chat completion endpoint.
+
+        :return: The response.
+        :rtype: dict
+        """
+        return openai.ChatCompletion.create(
+            engine="gpt-4-32k",
```
Comment (romanlutz) [inline python/ml_wrappers/model/openai_wrapper.py:50]: What about other types?

File: python/ml_wrappers/model/openai_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            messages=self.messages,
+            temperature=0.7,
+            max_tokens=800,
```
Comment (romanlutz) [inline python/ml_wrappers/model/openai_wrapper.py:53]: seems like we're making a lot of choices fixed here. Any reason why?

**Review by romanlutz — APPROVED** [review]

**Review by tongyu-microsoft — APPROVED** [review]

File: python/ml_wrappers/model/openai_wrapper.py (hunk 1 above)
Comment (imatiach-msft) [reply python/ml_wrappers/model/openai_wrapper.py]: good point, done, made this a parameter on the endpoint!

File: python/ml_wrappers/model/openai_wrapper.py (hunk 2 above)
Comment (imatiach-msft) [reply python/ml_wrappers/model/openai_wrapper.py]: done, made these parameters on the endpoint!

**Pull request merged**